source/runtime_gui.cpp

```diff
@@ -2168,8 +2173,8 @@ void reshade::runtime::draw_gui_settings()
 				"Block all input when overlay is visible\n");
 			std::replace(input_processing_mode_items.begin(), input_processing_mode_items.end(), '\n', '\0');
 			modified |= ImGui::Combo(_("Input processing"), reinterpret_cast<int *>(&_input_processing_mode), input_processing_mode_items.c_str());
-
-			modified |= imgui::key_input_box(_("Overlay key"), _overlay_key_data, *_input);
+            modified |= ImGui::Checkbox(_("Escape Closes Overlay"),&_esc_closes_overlay);
```

Comment:
I'm not convinced this needs to be a GUI option, since it's very niche seems sufficient to just be a config file option only.

Reply:
that's totally reasonable! my one concern is how to communicate that option to anyone (I'm sure I can't be the only one) who might have been frustrated by old behavior. I guess if its included in the changelog and shows up in the ini without having to be explicitly added that should be plenty. However in that case I do think it should stay as an input option rather than overlay. I would guess that anyone wanting this feature is using addons and has an overlay section a bit like below. Reason I say that is that REST already behaves like this whether intentional or not, which is also the reason I noticed the issue to begin with.

![image](https://github.com/user-attachments/assets/33f49da9-d420-48dd-aa60-65e5b9225100)

Personally I wouldn't be able to single out a new option in that section, here its easy to see and doesn't add to an already large section.

![image](https://github.com/user-attachments/assets/83d059ac-ca6a-4dc6-96c9-24169e4e0b15)

...Actually wait a second, the solution was already available in reshade, we just needed to make a behavior more consistent.

`if (_show_overlay && !_ignore_shortcuts && !_imgui_context->IO.NavVisible && _input->is_key_pressed(0x1B /* VK_ESCAPE */) &&
       ((_input_processing_mode == 2 || (_input_processing_mode == 1  &&  (_input->is_blocking_any_mouse_input() || _input->is_blocking_any_keyboard_input())))))	
        show_overlay = false;`
there's a shorter way to check if we're